src/nidigital/custom_types/history_ram_cycle_information.py

```diff
@@ -0,0 +1,45 @@
+class HistoryRAMCycleInformation(object):
+    def __init__(self, pattern_name, time_set_name, vector_number, cycle_number, scan_cycle_number, expected_pin_states, actual_pin_states, per_pin_pass_fail):
+        self.pattern_name = pattern_name
+        self.time_set_name = time_set_name
+        self.vector_number = vector_number
+        self.cycle_number = cycle_number
+        self.scan_cycle_number = scan_cycle_number
+        self.expected_pin_states = expected_pin_states
+        self.actual_pin_states = actual_pin_states
+        self.per_pin_pass_fail = per_pin_pass_fail
+
+    def __repr__(self):
+        parameter_list = [
+            'pattern_name="{}"'.format(self.pattern_name),
+            'time_set_name="{}"'.format(self.time_set_name),
+            'vector_number={}'.format(self.vector_number),
+            'cycle_number={}'.format(self.cycle_number),
+            'scan_cycle_number={}'.format(self.scan_cycle_number),
+            'expected_pin_states={}'.format(self.expected_pin_states),
+            'actual_pin_states={}'.format(self.actual_pin_states),
+            'per_pin_pass_fail={}'.format(self.per_pin_pass_fail),
+        ]
+
+        return '{0}({1})'.format(self.__class__.__name__, ', '.join(parameter_list))
+        return self.__class__.__name__
```

Comment:
The second return is redundant, I think.

Reply:
Yeah, that was a placeholder until I implemented the method body. Forgot to delete it later.